deeprankcore/DataSet.py

```diff
@@ -54,36 +54,58 @@ def save_hdf5_keys(
                 f_dest[key] = h5py.ExternalLink(f_src_path, "/" + key)
 
 
-def DivideDataSet(dataset, percent=None, shuffle=True):
+def _DivideDataSet(dataset, train_size=None):
     """Divides the dataset into a training set and an evaluation set
 
     Args:
-        dataset ([type])
-        percent (list, optional): [description]. Defaults to [0.8, 0.2].
-        shuffle (bool, optional): [description]. Defaults to True.
+        dataset (HDF5DataSet): input dataset to be split into training and validation data
+        train_size (float or int, optional): fraction of dataset (if float) or number of datapoints (if int) to use for training. 
+            Negative values will set size of the validation dataset instead.
+            Defaults to 0.75.
 
     Returns:
-        [type]: [description]
+        HDF5DataSet: [description]
     """
 
-    if percent is None:
-        percent = [0.8, 0.2]
-
-    size = len(dataset)
-    index = np.arange(size)
-
-    if shuffle:
-        np.random.shuffle(index)
-    size1 = int(percent[0] * size)
-    index1, index2 = index[:size1], index[size1:]
-
-    dataset1 = copy.deepcopy(dataset)
-    dataset1.index_complexes = [dataset.index_complexes[i] for i in index1]
-
-    dataset2 = copy.deepcopy(dataset)
-    dataset2.index_complexes = [dataset.index_complexes[i] for i in index2]
-
-    return dataset1, dataset2
+    full_size = len(dataset)
+
+    # find number of datapoints to include in training dataset
+    if train_size is None:
+        n_train = int(0.75 * full_size)
+    elif train_size is float:
+        n_train = int(train_size * full_size)
+    elif train_size is int:
+        if train_size == 1: # user probably intended it to be 1.0 (100%), not 1 datapoint
+            n_train = full_size
+            print ("WARNING: `train_size = 1` interpreted as 1.0 (100%), not as 1 datapoint")
+        else:
+            n_train = train_size
+        
+    # negative values means it is the number of datapoints to NOT include in training data
+    if n_train < 0:
+        n_train = full_size + n_train # adding negative number
+    
+    # raise exceptions if no training data or if more than 100% training data
+    if n_train > full_size:
+        raise ValueError ("invalid train_size: train_size too large. \n\t" +
+            f"train_size must be a float between -1 and 1 OR an int with max absolute value of len(dataset) ({full_size})")
+    if n_train <= 0:
```

Comment:
I would suggest removing the possibility to insert negative values. These are the constraints for `train_size` from sklearn `train_test_split` and I would stick with those: 
- Float: check that is between 0.0 and 1.0
- Int: check that is positive and <= total number of samples
- None: defaults to 0.75
They are basically already implemented, I would just add a check for float and a check for int, removing the negative value option

Reply:
The difference though is that in train_test_split you can set the train_size, test_size, or both (can be less than 100%). In that way you can just set the val_size directly. I figured to keep the number of parameters low, but still allow for the same functionality, it's nice to be able to use negative values.
I realize that it's a bit of overkill to allow all of these options, as they determine each other and there are different ways to get it done, but it does allow for different ways of thinking to work. For example, I would think more in terms of how much data I am willing to sacrifice as validation data, rather than in terms of how much I want to keep as training data. That's why I like being able to set it directly (also in train_test_split, it is primarily the test data that is set and train data that follows).

Also, in a way it makes sense that negative values determine the amount of data to NOT use for training, where positive values determine the amount of data to USE for training.

Reply:
> The difference though is that in train_test_split you can set the train_size, test_size, or both (can be less than 100%). In that way you can just set the val_size directly. I figured to keep the number of parameters low, but still allow for the same functionality, it's nice to be able to use negative values. I realize that it's a bit of overkill to allow all of these options, as they determine each other and there are different ways to get it done, but it does allow for different ways of thinking to work. For example, I would think more in terms of how much data I am willing to sacrifice as validation data, rather than in terms of how much I want to keep as training data. That's why I like being able to set it directly (also in train_test_split, it is primarily the test data that is set and train data that follows).
> 
> Also, in a way it makes sense that negative values determine the amount of data to NOT use for training, where positive values determine the amount of data to USE for training.

Also consider that this functionality is not supposed to be used a lot, it's just something that for now we're leaving there because it was already there and it could (very rarely) be handy. In the majority of cases, users are going to do the split before instantiating NeuralNet (and they do not want to randomly split data after shuffling; this is often done in ML tutorials, but in real use cases you want to have a clear distinction between which groups are in train and which ones are in validation, and this is especially true in our project domain). If you prefer we can have `test_size` instead of `train_size`, this would be fine for me.